lib/matplotlib/widgets.py

```diff
@@ -1049,6 +1078,14 @@ def set_active(self, index):
         for cid, func in self.observers.items():
             func(self.labels[index].get_text())
 
+    def clear(self):
+        """Deactivate any previously activated button."""
+        if self.index_selected is not None:
```

Comment:
My answer is also same as above that callbacks on `clear()` are counter-intuitive. Also, such a feature has to be thought of as common for all widgets, and hence its need needs to be discussed separately. Perhaps others @timhoffm @tacaswell can weigh in?

Reply:
I know you guys are extremely busy and this PR is not a high priority, but it would be great to check it off :).. Hi @anntzer , see my two replies here and above.

Reply:
There isn't a general clear() method for qt radio groups (http://doc.qt.io/qt-5/qbuttongroup.html), but at least there's a way to signal that a button was cleared (buttonReleased/buttonToggled).

Reply:
this is a matter of interpreting the existing attributes such as `value_selected` and `index_selected` - if they are `None`, it can be deduced the radio button is cleared (or not set). I don't think its necessary to create and expose new attributes to indicate a clear operation, which itself rare. Perhaps others @timhoffm @tacaswell can weigh in.

Reply:
I think the argument is rather that you do not want to have two differing paths in the code. I.e. if you have 3 checkboxes, the first one activated, and you deactivate this first box you end up with a "cleared" state. This would trigger any registered callback. If instead of deactivating the box, you `clear` it, you equally end up with a "cleared" state, but this time - as it stands now - no callback is triggered. This seems to be undesireable and might lead to confusion. Hence the proposal to call the callbacks on `clear` as well. This would ensure to go along the same code path, independent on how you clear the boxes.

Reply:
thanks, I see that. I am trying to point out that any callback execution - which are meant to execute only on well-defined states (what is provided by user) - on `clear()` would be unexpected to the user and hence undesirable to the user. If we implement it, we must take it as a separate argument during `init` and clearly label cleanup or callback on clear.. 

callbacks are NOT defined for `clear` state, unless we ask the user to ensure callbacks handle that too, which I believe would reduce the simplicity and usability of the current design. Are you following me?

i'd leave it to your judgement though, and implement them if you request it.

Reply:
This would really benefit from a concrete example of when the GUI programmer might want to call this method and why it should or should not execute call backs associated with the buttons/boxes.  Setting the appearance and state of a group of buttons without executing the callbacks seems strange to me.

Reply:
thanks guys - I've now implemented the callback execution even for `.clear()`. For some reason, can't run the tests locally on macbook, so we need to wait for the CI to pass.